src/pages/home/sidebar/SidebarLinks.js

```diff
@@ -176,9 +176,39 @@ export default compose(
         // with 10,000 withOnyx() connections, it would have unknown performance implications.
         reports: {
             key: ONYXKEYS.COLLECTION.REPORT,
+            selector: report => report && ({
+                reportID: report.reportID,
+                participants: report.participants,
+                hasDraft: report.hasDraft,
+                isPinned: report.isPinned,
+                errorFields: {
+                    addWorkspaceRoom: report.errorFields && report.errorFields.addWorkspaceRoom,
+                },
+                maxSequenceNumber: report.maxSequenceNumber,
+                lastReadSequenceNumber: report.lastReadSequenceNumber,
+                lastMessageText: report.lastMessageText,
+                lastMessageTimestamp: report.lastMessageTimestamp,
+                iouReportID: report.iouReportID,
+                hasOutstandingIOU: report.hasOutstandingIOU,
+                statusNum: report.statusNum,
+                stateNum: report.stateNum,
+                chatType: report.chatType,
+                policyID: report.policyID,
+            }),
```

Comment:
Having these large object literals in WithOnyx is kind of hard to read. I think we should put this outside withOnyx call as variables or functions either in the same file to a new file.

Reply:
I wouldn't be opposed to defining them directly above `withOnyx()`. Would that be OK?

Reply:
Yeah, that is totally fine too. There are no drawbacks as such.